src/css3_parser/tokenizer.rs

```diff
@@ -0,0 +1,158 @@
+use crate::css3_parser::tokens::{Token, TOKENS_REGEXS};
+use regex::{self, Regex};
+
+/// CSS Tokenizer
+#[derive(Debug, PartialEq)]
+pub struct CSSTokenizer {
+    pub cursor: usize,
+    raw: String,
+}
+
+impl Default for CSSTokenizer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CSSTokenizer {
+    pub fn new() -> Self {
+        CSSTokenizer {
+            cursor: 0,
+            raw: String::new(),
+        }
+    }
+    pub fn init(&mut self, raw: &str) {
+        self.raw = raw.to_string();
+        self.cursor = 0;
+    }
+
+    pub fn has_more_tokens(&self) -> bool {
+        self.cursor < self.raw.len()
+    }
+
+    pub fn is_eof(&self) -> bool {
+        self.cursor == self.raw.len()
+    }
+
+    pub fn get_next_token(&mut self) -> Option<Token> {
+        if !self.has_more_tokens() {
+            return None;
+        }
+
+        let raw = &self.raw[self.cursor..];
+        for (regex, token_type) in TOKENS_REGEXS.iter() {
```

Comment:
I'm guessing the use of regular expressions in the inner loop parser to identify the tokens will need to be provisional, for performance reasons.  But that can be profiled later and addressed if needed.